modules/masterbar/tracks-events.js

```diff
@@ -69,5 +44,65 @@
 		},
 	};
 
-	window.wpcom_masterbar.user_id = jetpackTracks.user_id;
-})( jetpackTracks );
+	var nonce = jetpackTracks.tracks_nonce;
+
+	function parseJson( s, defaultValue ) {
+		try {
+			return JSON.parse( s );
+		} catch ( e ) {
+			return defaultValue;
+		}
+	}
+
+	$( document ).ready( function(){
+		$( '.ab-item, .ab-secondary' ).on( 'click touchstart', function( e ) {
+			var $target = $( e.target ),
+					$parent = $target.closest( 'li' );
+
+			if( ! $parent ) {
+				return;
+			}
+
+			var parentId = $parent.attr( 'ID' );
+			var trackId = $target.hasClass( 'ab-secondary' ) ? parentId + '-secondary' : parentId;
+			var eventName = linksTracksEvents[ trackId ] || linksWhitelist.indexOf( trackId ) || null;
+			if ( ! eventName ) {
+				return;
+			}
+
+			if( $parent.hasClass( 'menupop' ) ) {
+				//top level items that open a panel
+				window._tkq.push( [ 'recordEvent', eventName ] );
+			} else {
+				e.preventDefault();
+				window.location = 'jetpack-track-and-bounce.php?' + $.param( {
+					tracks_and_bounce: trackId,
+					tracks_and_bounce_nonce: nonce,
+				} );
+			}
+		} );
+	} );
+
+	// listen for postMessage events from the notifications iframe
+	$( window ).on( 'message', function( e ) {
+		var event = ! e.data && e.originalEvent.data ? e.originalEvent : event;
+		if ( event.origin !== 'https://widgets.wp.com' ) {
+			return;
+		}
+
+		var data = ( 'string' == typeof event.data ) ? parseJson( event.data, {} ) : event.data;
+		if ( 'notesIframeMessage' !== data.type ) {
+			return;
+		}
+
+		var eventData = notesTracksEvents[ data.action ];
+		if ( ! eventData ) {
+			return;
+		}
+
+		window._tkq.push( [ 'recordEvent', eventData.name, eventData.properties( data ) ] );
+	} );
+
+	window.jetpackTracks = null;
```

Comment:
why set this to null here?

Reply:
the intention is to remove the _nonce_ from the global scope so it can't be used by other functions.